spec/validation_spec.rb

```diff
@@ -1,10 +1,33 @@
 require 'nokogiri'
 require 'pp'
+require 'open-uri'
 
 
 RSpec.describe 'Validate Examples' do
   before :all do
-    @xsd = Nokogiri::XML::Schema(File.read('BuildingSync.xsd'))
+    # Nokogiri doesn't seem to support XSDs which import other schemas with URLs
+    # for the schemaLocation. To allow testing, we download the imported schema
+    # and point schemaLocation to it instead
+    schema_doc = Nokogiri::XML(File.read('BuildingSync.xsd'))
+
+    GEOJSON_XSD_PATH = 'geojson.xsd'
+    GEOJSON_IMPORT_XPATH = 'xs:schema/xs:import[@namespace = "http://www.gbxml.org/schema"]'
+    if !File.file?(GEOJSON_XSD_PATH) then
```

Comment:
thanks for adding this hack!

Reply:
For backend code, where an XSD file is used, but the end-user never sees the XSD file directly (such as the `BuildingSync.xsd` file in Audit Template's backend, which is used by the APIs for validation), I like to cache local copies of the XSD file and its `xs:import`s (and the imports of the imports, etc.), and then to change all of the `schemaLocation` attribute values from URLs to paths.

For example, in Audit Template's `xsd2ruby.rb` code generation pipeline, the gbXML import would be changed from

```
<xs:import namespace="http://www.gbxml.org/schema" schemaLocation="https://www.gbxml.org/schema/6-01/GreenBuildingXML_Ver6.01.xsd"/>
```

to

```
<xs:import namespace="http://www.gbxml.org/schema" schemaLocation="./GreenBuildingXML_Ver6.01.xsd"/>
```

Reply:
Also, note the weird "file:///F:/Temp/GreenBuildingXML_Ver6.01.xsd" value for the `schemaLocation` attribute in the `GreenBuildingXML_Ver6.01.xsd` file.